crates/flipt/src/flipt.rs

```diff
@@ -0,0 +1,195 @@
+where
+    A: flipt::AuthenticationStrategy,
+{
+    pub url: String,
+    pub authentication_strategy: A,
+    pub timeout: u64,
+}
+
+pub struct FliptProvider {
+    metadata: ProviderMetadata,
+    client: flipt::api::FliptClient,
+    namespace: String,
+}
+
+impl FliptProvider {
+    pub fn new<A: flipt::AuthenticationStrategy>(
+        namespace: String,
+        config: Config<A>,
+    ) -> Result<Self, String> {
+        let url = match Url::parse(&config.url) {
+            Ok(url) => url,
+            Err(e) => return Err(e.to_string()),
+        };
+
+        let flipt_config = flipt::Config::new(url, config.authentication_strategy, config.timeout);
+        let client = match flipt::api::FliptClient::new(flipt_config) {
+            Ok(fpconfig) => fpconfig,
+            Err(e) => return Err(e.to_string()),
+        };
+
+        Ok(Self {
+            metadata: ProviderMetadata::new(METADATA),
+            client,
+            namespace,
+        })
+    }
+}
+
+#[async_trait]
+impl FeatureProvider for FliptProvider {
+    fn metadata(&self) -> &ProviderMetadata {
+        &self.metadata
+    }
+
+    async fn resolve_bool_value(
+        &self,
+        flag_key: &str,
+        ctx: &EvaluationContext,
+    ) -> EvaluationResult<ResolutionDetails<bool>> {
+        let res = match self
+            .client
+            .evaluation
+            .boolean(&EvaluationRequest {
+                namespace_key: self.namespace.clone(),
+                flag_key: flag_key.into(),
+                entity_id: ctx
+                    .targeting_key
+                    .clone()
+                    .unwrap_or(DEFAULT_ENTITY_ID.to_owned()),
+                context: translate_context(ctx),
+                reference: None,
+            })
+            .await
+        {
+            Ok(r) => r,
+            Err(e) => return Err(translate_error(e)),
+        };
+
+        EvaluationResult::Ok(ResolutionDetails::new(res.enabled))
+    }
+
+    async fn resolve_int_value(
+        &self,
+        flag_key: &str,
+        ctx: &EvaluationContext,
+    ) -> EvaluationResult<ResolutionDetails<i64>> {
+        let res = match variant_helper(self, flag_key, ctx).await {
+            Ok(r) => r,
+            Err(e) => return Err(e),
+        };
+        // parse a variant key as i64
+        res.variant_key
+            .parse::<i64>()
+            .map_err(|e| EvaluationError {
+                code: EvaluationErrorCode::General("Parse error".to_owned()),
+                message: Some(format!(
+                    "Expected a number in range of i64, but found `{}` ({:?})",
+                    res.variant_attachment, e
+                )),
+            })
+            .map(|v| ResolutionDetails::new(v))
+    }
+
+    async fn resolve_float_value(
+        &self,
+        flag_key: &str,
+        ctx: &EvaluationContext,
+    ) -> EvaluationResult<ResolutionDetails<f64>> {
+        let res = match variant_helper(self, flag_key, ctx).await {
```

Comment:
It appears we can write it as follows.
```suggestion
        let res = match variant_helper(self, flag_key, ctx).await?;
```